sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1727,6 +1727,91 @@ __ESIMD_API __ESIMD_NS::simd<T, N> dpasw2(
 }
 /// @} sycl_esimd_systolic_array_api
 
+/// @addtogroup sycl_esimd_logical
+/// @{
+
+/// Performs binary function computation with three vector operands.
```

Comment:
I may be wrong, but I think this comment should be placed right before the bfn() function definition.
Otherwise, this comment would stick to 'enum class bfn_t'.

Reply:
Surely, the enum defined below needs some doxy-gen comment.
The lines with `tparam` and 'param' looks a bit off here.

I would just write something like:L
```suggestion
/// This enum is used to encode all possible combinations of logical
/// operations performed on 3 input operand. It is used as a template
/// argument of bfn() function.
/// Example: d = bfn<~bfn::x & ~bfn::y & ~bfn::z>(s0, s1, s2);
```